pkg/reconciler/instances/istio/reset/pod/handler.go

```diff
@@ -3,30 +3,49 @@ package pod
 import (
 	"context"
 	"fmt"
-	"sync"
 	"time"
 
 	"github.com/avast/retry-go"
-	"k8s.io/client-go/kubernetes"
-
 	"go.uber.org/zap"
+	v1 "k8s.io/api/apps/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	kctlutil "k8s.io/kubectl/pkg/util/deployment"
 )
 
-type GetSyncWG func() *sync.WaitGroup
-
 //go:generate mockery --name=Handler --outpkg=mocks --case=underscore
-// Handler executes actions on the Kubernetes cluster
+// Handler executes actions on the Kubernetes objects.
 type Handler interface {
-	Execute(CustomObject, GetSyncWG)
+	// Execute action on the Kubernetes object with regards of the type of handler.
+	// Returns error if action was unsuccessful or wait timeout was reached.
+	ExecuteAndWaitFor(CustomObject) error
```

Comment:
We have already an implememtation for such wait-actions. Please see example implementation here:

https://github.com/kyma-incubator/reconciler/blob/main/pkg/reconciler/kubernetes/progress/tracker_test.go#L110-L118

Reply:
I guess both of them were created at the same time, I have added an issue about the follow-up.
https://github.com/kyma-incubator/reconciler/issues/369